cli.js

```diff
@@ -31,6 +32,7 @@ const cli = meow(`
 	  --stdin           Validate/fix code from stdin
 	  --stdin-filename  Specify a filename for the --stdin option
 	  --print-config    Print the effective ESLint config for the given file
+		--flat            Use the experimental flat config.
```

Comment:
If already reliable, I think it's best to switch to the flat config rather than add another thing to maintain. There's no advantage for the user to run this experiment I think. 

As a tester I'd probably prefer trying a flat-only version published from this branch, rather than something that will eventually require further refactoring and further testing. 

_Note: I'm not a maintainer here so this is just an opinion_

Reply:
@fregante the problem right now is that the flat config is not currently reliable (on eslints side) the api will change, and I had to implement a couple of hacky work arounds until eslint gets it stable. 

The benefit to the user right now is that they can use third party plugins and override ours without resolution problems. So there would be benefits to using it right now if you want to add or remove plugins and third party configs or support something like mdx or whatever else eslint can support, which is impossible (or difficult and bug prone) with xo today.